### PR TITLE
Add "Trivy" security check to Build Docker Images workflow

### PR DESCRIPTION
**Pull request by mindaugasrukas**
## What was changed

Created reusable GitHub action for Trivy security checks.
Added the "Trivy" check to the Build Docker Images workflow.

More info https://github.com/aquasecurity/trivy-action

## Why?

To get earlier feedback about security issues.

## Checklist

1. Closes 
No
2. How was this tested:
CI
3. Any docs updates needed?
No

### CODE REVIEW

File: .github/actions/trivy/action.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      id: vars
+      shell: bash
+      run: |
+        tag=$(echo "$TAGS" | head -1)
+        name=${NAME// /-}
+        echo "::set-output name=tag::${tag}"
+        echo "::set-output name=name::${name}"
+
+    - name: Run Trivy vulnerability scanner on ${{ inputs.image-name }} image
+      uses: aquasecurity/trivy-action@master
```
Comment (dnr) [inline .github/actions/trivy/action.yml:28]: do they really recommend using master? or should it be some kind of version tag?

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -85,6 +91,13 @@ jobs:
           build-args: |
             SERVER_IMAGE=temporaliotest/server:${{ steps.build_args.outputs.image_tag }}
 
+      - name: Run Trivy vulnerability scanner on Admin Tools image
+        if: steps.build_args.outputs.push == 'true'
+        uses: ./.github/actions/trivy
+        with:
+          image-tags: ${{ steps.meta_admin_tools.outputs.tags }}
+          image-name: admin_tools
```
Comment (dnr) [inline .github/workflows/docker.yml:99]: isn't it `admin-tools`?

File: .github/workflows/docker.yml [hunk 3]
```diff
@@ -125,3 +138,10 @@ jobs:
           build-args: |
             SERVER_IMAGE=temporaliotest/server:${{ steps.build_args.outputs.image_tag }}
             ADMIN_TOOLS_IMAGE=temporaliotest/admin-tools:${{ steps.build_args.outputs.image_tag }}
+
+      - name: Run Trivy vulnerability scanner on Auto Setup image
+        if: steps.build_args.outputs.push == 'true'
+        uses: ./.github/actions/trivy
+        with:
+          image-tags: ${{ steps.meta_auto_setup.outputs.tags }}
+          image-name: auto_setup
```
Comment (dnr) [inline .github/workflows/docker.yml:147]: same

File: .github/workflows/docker.yml [hunk 4]
```diff
@@ -85,6 +91,13 @@ jobs:
           build-args: |
             SERVER_IMAGE=temporaliotest/server:${{ steps.build_args.outputs.image_tag }}
 
+      - name: Run Trivy vulnerability scanner on Admin Tools image
```
Comment (dnr) [inline .github/workflows/docker.yml:94]: are we going to see triple warnings for things that are present in all three images?

File: .github/workflows/docker.yml [hunk 5]
```diff
@@ -60,6 +60,12 @@ jobs:
           tags: ${{ steps.meta_server.outputs.tags }}
           labels: ${{ steps.meta_server.outputs.labels }}
 
+      - name: Run Trivy vulnerability scanner on Server image
+        uses: ./.github/actions/trivy
```
Comment (dnr) [inline .github/workflows/docker.yml:64]: does this block PRs if it finds things? if it doesn't, and we want it to, what would change here? (or if it does, and we don't want it to, ...?)

File: .github/actions/trivy/action.yml (hunk 1 above)
Comment (mindaugasrukas) [reply .github/actions/trivy/action.yml:28]: Good question. This is based on their documentation. I guess we want to use the latest version always.

File: .github/workflows/docker.yml (hunk 4 above)
Comment (mindaugasrukas) [reply .github/workflows/docker.yml:94]: Yes, as I understand, all three scan results will be available.

File: .github/workflows/docker.yml (hunk 2 above)
Comment (mindaugasrukas) [reply .github/workflows/docker.yml:99]: This name doesn't have any meaning except logging. I don't have any preferences here. I can change the formatting.

File: .github/workflows/docker.yml (hunk 5 above)
Comment (mindaugasrukas) [reply .github/workflows/docker.yml:64]: Currently, it fails PRs but allows us to merge anyways. We can tune this as we like.

Comment (dnr) [reply .github/workflows/docker.yml:64]: yeah, I was just curious what the change to block merging would look like

File: .github/workflows/docker.yml (hunk 2 above)
Comment (dnr) [reply .github/workflows/docker.yml]: if it doesn't have any meaning, how does the action know what image to scan? is `tags` a full name including repository and tag?

**Review by dnr — APPROVED** [review]

File: .github/workflows/docker.yml (hunk 2 above)
Comment (mindaugasrukas) [reply .github/workflows/docker.yml]: Yes, `tags` contain full image names, for example:
```
temporaliotest/server:latest
temporaliotest/server:sha-24fcf99
```

**Pull request merged**